2018-pycon-russia/MLonCode.md

```diff
@@ -0,0 +1,28 @@
+# Machine Learning on Source Code
+
+**Updates**:  https://github.com/src-d/conferences/issues/49
+**Sent on**:  2018/03/20
+**Status**:   submitted
+**Author**:   Vadim Markovtsev
+**Slides**:   TBA
+**Proposal**: N/A
+
+**Abstract**:
+
+Machine Learning on Source Code (MLoSC) is an emerging and exciting domain of research which stands at the sweet spot between deep learning, natural language processing, social science and programming. The list of MLoSC resources - [awesome-machine-learning-on-source-code](https://github.com/src-d/awesome-machine-learning-on-source-code) - has already attracted more than 1,600 watchers on GitHub.
+
+I will summarize the current research subdomains in MLoSC, such as *identifier and structural embeddings*, *programming language modeling*, *automated programming language evolution* and *source code topic modeling and exploratory search*. I will also describe the open MLoSC toolbox from [source{d}](https://sourced.tech) based on PySpark: [sourced.engine](https://github.com/src-d/engine) and [sourced.ml](https://github.com/src-d/ml). It provides an in-depth set of Python APIs and command line applications which is already being used for various production tasks. Finally, I will show how to apply that toolbox to analyze all world's Python open source code and mine gems.
```

Comment:
I thought `gems` was a Ruby term, not Python?

Reply:
Hehe. These are the real gems which you mine. Python has wheels and eggs.